File: Backend (Springboot)/Annapurna/src/main/java/com/annapurna/service/RazorpayService.java
package com.annapurna.service;

import com.razorpay.Order;
import com.razorpay.RazorpayClient;
import com.razorpay.RazorpayException;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
public class RazorpayService {

    private final RazorpayClient razorpayClient;

    public RazorpayService(@Value("${razorpay.key_id}") String keyId,
                           @Value("${razorpay.key_secret}") String keySecret) throws RazorpayException {
        this.razorpayClient = new RazorpayClient(keyId, keySecret);
    }

    public Order createOrder(int amountInPaise, String currency, String receipt) throws RazorpayException {
        JSONObject orderRequest = new JSONObject();
        orderRequest.put("amount", amountInPaise); // Already in paise
        orderRequest.put("currency", currency);
        orderRequest.put("receipt", receipt);
        orderRequest.put("payment_capture", 1); // Auto capture

        return razorpayClient.orders.create(orderRequest);
    }
}

File: Backend (DNET)/Annapurna/Services/RazorpayService.cs
using Razorpay.Api;
using System.Collections.Generic;

namespace AnnapurnaAPI.Services
{
    public class RazorpayService
    {
        private readonly RazorpayClient _client;

        public RazorpayService(IConfiguration configuration)
        {
            var keyId = configuration["Razorpay:KeyId"];
            var keySecret = configuration["Razorpay:KeySecret"];
            _client = new RazorpayClient(keyId, keySecret);
        }

        public Dictionary<string, object> CreateOrder(long amount, string currency, string receipt)
        {
            var options = new Dictionary<string, object>
            {
                { "amount", amount },
                { "currency", currency },
                { "receipt", receipt }
            };

            var order = _client.Order.Create(options);
            return order.Attributes;
        }
    }
}
